mne/preprocessing/maxwell.py

```diff
@@ -0,0 +1,526 @@
+
+    return int_order ** 2 + 2 * int_order + ext_order ** 2 + 2 * ext_order
+
+
+def _sph_to_cart_partials(sph_pts, sph_grads):
+    """Convert spherical partial derivatives to cartesian coords.
+
+    Note: Because we are dealing with partial derivatives, this calculation is
+    not a static transformation. The transformation matrix itself is dependent
+    on azimuth and polar coord.
+
+    See the 'Spherical coordinate sytem' section here:
+    wikipedia.org/wiki/Vector_fields_in_cylindrical_and_spherical_coordinates
+
+    Parameters
+    ----------
+    sph_pts : ndarray, shape (n_points, 3)
+        Array containing spherical coordinates points (rad, azimuth, polar)
+    sph_grads : ndarray, shape (n_points, 3)
+        Array containing partial derivatives at each spherical coordinate
+
+    Returns
+    -------
+    cart_grads : ndarray, shape (n_points, 3)
+        Array containing partial derivatives in Cartesian coordinates (x, y, z)
+    """
+
+    cart_grads = np.zeros_like(sph_grads)
+
+    # TODO: needs vectorization, currently matching Jussi's code for debugging
+    for pt_i, (sph_pt, sph_grad) in enumerate(zip(sph_pts, sph_grads)):
+        # Calculate cosine and sine of azimuth and polar coord
+        c_a, s_a = np.cos(sph_pt[1]), np.sin(sph_pt[1])
+        c_p, s_p = np.cos(sph_pt[2]), np.sin(sph_pt[2])
+
+        trans = np.array([[c_a * s_p, -s_a, c_a * c_p],
+                          [s_a * s_p, c_a, c_p * s_a],
+                          [c_p, 0, -s_p]])
+
+        cart_grads[pt_i, :] = np.dot(trans, sph_grad)
+
+    return cart_grads
+
+
+def _cart_to_sph(cart_pts):
+    """Convert Cartesian coordinates to spherical coordinates.
+
+    Parameters
+    ----------
+    cart_pts : ndarray, shape (n_points, 3)
+        Array containing points in Cartesian coordinates (x, y, z)
+
+    Returns
+    -------
+    sph_pts : ndarray, shape (n_points, 3)
+        Array containing points in spherical coordinates (rad, azimuth, polar)
+    """
+
+    rad = np.sqrt(np.sum(cart_pts * cart_pts, axis=1))
+    az = np.arctan2(cart_pts[:, 1], cart_pts[:, 0])
+    pol = np.arccos(cart_pts[:, 2] / rad)
+
+    return np.c_[rad, az, pol]
+
+
+# TODO: Eventually refactor this in forward computation code
+
+def _make_coils(info, accurate=True, elekta_defs=False):
+    """Prepare dict of MEG coils and their information.
+
+    Parameters
+    ----------
+    info : instance of mne.io.meas_info.Info
+        Info dict from Raw FIF file
+    accurate : bool
+        If True, use high accuracy of coil information (more integration
+        points)
+    elekta_defs : bool
+        If True, use Elekta's integration point geometry
+
+    Returns
+    -------
+    megcoils, meg_info : dict
+        MEG coils and information dict
+    """
+
+    if accurate:
+        accuracy = FIFF.FWD_COIL_ACCURACY_ACCURATE
+    else:
+        accuracy = FIFF.FWD_COIL_ACCURACY_NORMAL
+    meg_info = None
+    megcoils = []
```

Comment:
for consistency with extended MNE style guidelines put list() here instead of [ ]